### PR TITLE
Issue #6560: run tests with English locale and UTF-8 encoding by default

### LINKED ISSUE #6560: Tests fail with German locale on Windows with default encoding
Followup of #6511

CS tests fail with German locale on Windows with default encoding due to encoding problems.
CI with German locale is most probably passing because it is run on Linux where default encoding is `UTF-8`.

When I run `mvn clean test -DargLine='-Duser.language=de -Duser.country=DE -Dfile.encoding=Cp1252' -Dtest=com.puppycrawl.tools.checkstyle.DefaultLoggerTest#testAddError` on the latest master, I get errors like:
grafik

Using the checkstyle all jar, it always outputs in `UTF-8` which might be fine in a generated file, but is bad on the commandline.
grafik
(the prompt is Cygwin, but looks the same in `cmd.exe`, I just tried)

Default encoding for Windows in most western european locales is Cp1252, so is it here.

I propose to run the tests by default with `en_US.UTF-8` locale, for better build reproducability and test stability.

### PR DESCRIPTION
**Pull request by Vampire**
Fixes #6560

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -212,6 +212,7 @@
     <pitest.plugin.timeout.constant>50000</pitest.plugin.timeout.constant>
     <pitest.plugin.threads>4</pitest.plugin.threads>
     <sonar.test.exclusions>**/test/resources/**/*,**/it/resources/**/*</sonar.test.exclusions>
+    <argLine>-Duser.language=en -Duser.country=US -Dfile.encoding=UTF-8</argLine>
```
Comment (romani) [inline pom.xml]: we spend some much time to make localized builds, I am not ok to make `-Duser.language=en -Duser.country=US` so easily.
I am ok `-Dfile.encoding=UTF-8` 

I might be ok with extra maven profiles to activate on windows only when encoding is not UTF-8.

Comment (rnveach) [reply pom.xml]: @romani issue created identified the underlying issue, which is an issue with `OutputStreamWriter` and I think we need to convert it to `PrintWriter`.

Comment (romani) [reply pom.xml]: if problems in code, it should be fixed in code, but not in configuration for all OSs.

Comment (rnveach) [reply pom.xml]: I agree which is why I dug up the underlying issue. I can fix up the issue to make it clearer.

**Comment by romani** [comment]
@Vampire , am closing this PR, let's finish discussion in issue first. In worth case I think we can do something in maven with maven profiles to make it work.

**Pull request closed without merging**